About
=====

----------------------------
The MGnify service
----------------------------

MGnify is a freely available hub for the analysis and exploration of :term:`metagenomic`, :term:`metatranscriptomic`, :term:`amplicon` and :term:`assembly` data. The resource provides rich functional and taxonomic analyses of user-submitted sequences, as well as analysis of publicly available metagenomic datasets held within the European Nucleotide Archive (`ENA <https://www.ebi.ac.uk/ena>`_).
MGnify resource was previously known as EBI Metagenomics, but was rebranded to reflect its growing remit.

--------------------------------
What does MGnify offer
--------------------------------
**Standards-compliant data**
    The service provides submission tools that help ensure sequence data and metadata comply with the European Nucleotide Archive (ENA) data schemes and the Genomic Standards Consortium (`GSC <http://gensc.org>`_) metadata guidelines, allowing harmonisation of analysis efforts across the wider genomics community.
**Powerful analysis**
    The service provides powerful taxonomic and functional analysis of sequence data. Functional analysis results can be summarized and compared using Gene Ontology (`GO <http://www.geneontology.org>`_) terms.
**Data Archiving**
    Data submitted to the service is automatically archived in the ENA. Accession numbers are supplied for sequence data as part of the archiving process, which is a prerequisite for publication in many journals.
**Programmatic access to the data**
    Data deposited in the service are discoverable and queryable via RESTful API using unique URLs constructed from accession numbers supplied during archiving process.

-------
Funding
-------
MGnify was initiated with funding from EMBL. It continues to be developed with EMBL support and funding gratefully received from the Biotechnology and Biological Sciences Research Council (BBSRC grant BB/M011755/1) and [ELIXIR GRANT INFO]. It previously received funding from BBSRC grant BB/I02612X/1 and the EU's Seventh Framework Programme for Research (FP7 grant MICROB3).


-----------
How to cite
-----------
If you find this resource useful, please consider citing the following publication:

| `EBI Metagenomics in 2017: enriching the analysis of microbial communities, from sequence reads to assemblies. <https://academic.oup.com/nar/article/46/D1/D726/4561650>`_
| A Mitchell, M Scheremetjew, H Denise, S Potter, A Tarkowska, M Qureshi, G A Salazar, S Pesseat, M A Boland, F M I Hunter, P ten Hoopen, B Alako, C Amid, D J Wilkinson, T P Curtis, G Cochrane, R D Finn.
| **Nucleic Acids Research** (2017) Database Issue 46:D726â€“D735. `10.1093/nar/gkx967 (PDF) <https://academic.oup.com/nar/article-pdf/46/D1/D726/23162872/gkx967.pdf>`_

-------
Contact
-------
If you would like to get in touch, please contact us using the `helpdesk <metagenomics-help@ebi.ac.uk>`_. We will respond as quickly as possible, but please bear in mind that we do not have a member of staff dedicated to running the helpdesk service.

   **Please, check our** :ref:`FAQ <faq>` **prior to contacting us.**

If you would like to keep up to date with developments with the MGnify, please follow us on Twitter (`@EBImetagenomics <https://twitter.com/ebimetagenomics>`_).

